imessage-exporter/src/exporters/html.rs

```diff
@@ -399,12 +401,18 @@ impl<'a> Writer<'a> for HTML<'a> {
         match attachment.path() {
             Some(path) => {
                 if let Some(path_str) = path.as_os_str().to_str() {
-                    // Resolve the attachment path if necessary
-                    let resolved_attachment_path = if path.starts_with("~") {
-                        path_str.replace('~', &home())
-                    } else {
-                        path_str.to_owned()
-                    };
+                    let resolved_attachment_path = 
+                        // Resolve the relative attachment path to absolute for macOS
+                        if path.starts_with("~") & !self.config.options.ios {
+                            path_str.replace('~', &home())
+                        // Resolve the attachment path for iOS backup
+                        } else if self.config.options.ios {
+                            let salt = "MediaDomain-";
+                            let hash = format!("{:x}", Sha1::digest(format!("{}{}", salt, &path_str[2..]).as_bytes()));
```

Comment:
Use https://doc.rust-lang.org/std/vec/struct.Vec.html#method.get, don't directly index as it can panic.